### PR TITLE
Apple Sign In support, wallet improvements.

### PR DESCRIPTION
**Pull request by zyro**

### CODE REVIEW

File: apigrpc/apigrpc.swagger.json [hunk 1]
```diff
@@ -100,21 +133,6 @@
             "schema": {
               "$ref": "#/definitions/apiAccountCustom"
             }
-          },
```
Comment (novabyte) [inline apigrpc/apigrpc.swagger.json:103]: @zyro You'll need to manually undo this changes the omit query params in POST and DELETE methods because they're not generated correctly by Swagger codegen so we manually patch them in.

File: server/console_account.go [hunk 2]
```diff
@@ -505,31 +505,3 @@ AND ((facebook_id IS NOT NULL
 
 	return &empty.Empty{}, nil
 }
-
-func checkWalletFormat(wallet map[string]interface{}, path string) error {
```
Comment (novabyte) [inline server/console_account.go:509]: @zyro We should discuss this change. I'll run it through with you before we merge this pull request.

File: server/console_unlink.go [hunk 3]
```diff
@@ -24,6 +24,37 @@ import (
...
+	userID, err := uuid.FromString(in.Id)
+	if err != nil {
+		return nil, status.Error(codes.InvalidArgument, "Requires a valid user ID.")
+	}
+
+	query := `UPDATE users SET apple_id = NULL, update_time = now()
+WHERE id = $1
+AND apple_id IS NOT NULL
+AND ((custom_id IS NOT NULL
+			OR facebook_id IS NOT NULL
```
Comment (novabyte) [inline server/console_unlink.go:37]: nit: whitespace

File: social/social.go [hunk 4]
```diff
@@ -500,6 +528,148 @@ func (c *Client) GetSteamProfile(ctx context.Context, publisherKey string, appID
 	return profileWrapper.Response.Params, nil
 }
 
+func (c *Client) CheckAppleToken(ctx context.Context, bundleId string, idToken string) (*AppleProfile, error) {
```
Comment (novabyte) [inline social/social.go:531]: We should dump the `idToken` into DEBUG level logs to help diagnose any issues with the input sent by game clients.

File: apigrpc/apigrpc.swagger.json (hunk 1 above)
Comment (zyro) [reply apigrpc/apigrpc.swagger.json:103]: Looks like I reverted most but not all, will fix.

File: server/console_account.go (hunk 2 above)
Comment (zyro) [reply server/console_account.go:509]: Sounds good. :+1:

File: server/console_unlink.go (hunk 3 above)
Comment (zyro) [reply server/console_unlink.go:37]: Fixed.

File: social/social.go (hunk 4 above)
Comment (zyro) [reply social/social.go:531]: Doable, but we should talk this through first.

**Comment by lugehorsam** [comment]
I made the following issue to track doc changes: https://github.com/heroiclabs/nakama-docs/issues/152

**Comment by lugehorsam** [comment]
@zyro did this PR wind up getting merged elsewhere? Was going to do a documentation/sdk pass but looks like this PR is closed without being merged.

**Comment by zyro** [comment]
@lugehorsam Merged in commit https://github.com/heroiclabs/nakama/commit/4b92d9fb24cd281bf6de00e0c1d3ab82f22029ba.

**Comment by lugehorsam** [comment]
Gotcha. Must be a Squash and Merge quirk that Github says its close with unmerged commits.
